src/pages/Dashboard/Modals/SupplyWithdraw/index.tsx

```diff
@@ -218,62 +222,96 @@ const SupplyWithdrawModal: React.FC<ISupplyWithdrawUiProps> = props => {
       },
     },
   );
-  const { mutate: redeemUnderlying, isLoading: isRedeemUnderlyingLoading } = useRedeemUnderlying(
-    {
-      assetId: asset?.id as VTokenId,
-      account: account?.address || '',
-    },
-    {
-      onError: () => {
-        toast.error({
-          title: t('supplyWithdraw.withdrawError.title'),
-          description: t('supplyWithdraw.withdrawError.description', { symbol: asset.symbol }),
-        });
+  const { mutateAsync: redeemUnderlying, isLoading: isRedeemUnderlyingLoading } =
+    useRedeemUnderlying(
+      {
+        assetId: asset?.id as VTokenId,
+        account: accountAddress,
       },
-    },
-  );
+      {
+        onError: () => {
+          toast.error({
+            title: t('supplyWithdraw.withdrawError.title'),
+            description: t('supplyWithdraw.withdrawError.description', { symbol: asset.symbol }),
+          });
+        },
+      },
+    );
   const isWithdrawLoading = isRedeemLoading || isRedeemUnderlyingLoading;
-  const onSubmitSupply: IAmountFormProps['onSubmit'] = value => {
-    supply({
-      amount: getBigNumber(value)
+  const onSubmitSupply: IAmountFormProps['onSubmit'] = async value => {
+    try {
+      const withdrawableAmount = new BigNumber(value)
         .times(new BigNumber(10).pow(asset.decimals || 18))
-        .toString(10),
-    });
+        .toString(10);
+      const res = await supply({
```

Comment:
If `supply` throws an error, two error toasts will display: one from the `catch` block here and one from the `onError` callback. Same thing with the `supply` function.

Also we should follow the same pattern we've agreed upon and handle all UI-related actions from the UI component.